scene/resources/camera_texture.cpp

```diff
@@ -82,13 +86,26 @@ RID CameraTexture::get_rid() const {
 }
 
 Ref<Image> CameraTexture::get_image() const {
-	// not (yet) supported
-	return Ref<Image>();
+	return RenderingServer::get_singleton()->texture_2d_get(get_rid());
 }
 
 void CameraTexture::set_camera_feed_id(int p_new_id) {
+	Ref<CameraFeed> feed = CameraServer::get_singleton()->get_feed_by_id(camera_feed_id);
+	if (feed.is_valid()) {
+		if (feed->is_connected("format_changed", callable_mp(this, &CameraTexture::_on_format_changed))) {
+			feed->disconnect("format_changed", callable_mp(this, &CameraTexture::_on_format_changed));
+		}
+	}
+
 	camera_feed_id = p_new_id;
+
+	feed = CameraServer::get_singleton()->get_feed_by_id(camera_feed_id);
+	if (feed.is_valid()) {
+		feed->connect("format_changed", callable_mp(this, &CameraTexture::_on_format_changed));
+	}
+
 	notify_property_list_changed();
+	call_deferred(SNAME("emit_changed"));
```

Comment:
Isn't `callable_mp((Resource *)this, &Resource::emit_changed).call_deferred();` the safer way? In this case we can call `_on_format_changed` here directly (maybe rename that function even).